### PR TITLE
[RS] Fix ROOT-2558, SPlot cannot deal with ratios.

### PR DESCRIPTION
**Pull request by hageboeck**
Add the possibility to use ratios as yield parameters in SPlot.
Also update the documentation of SPlot, and add a test.

### CODE REVIEW

File: roofit/roostats/src/SPlot.cxx [hunk 1]
```diff
@@ -151,19 +202,20 @@ SPlot::SPlot(const char* name, const char* title, RooDataSet& data, RooAbsPdf* p
...
-   if(cloneData == 1) {
+  if(cloneData == 1) {
     fSData = (RooDataSet*) data.Clone(newName);
     SetBit(kOwnData);
-   }
+  }
   else
     fSData = (RooDataSet*) &data;
 
   // Add check that yieldsList contains all RooRealVars
```
Comment (eguiraud) [inline roofit/roostats/src/SPlot.cxx:212]: comment is outdated?

File: roofit/roostats/src/SPlot.cxx [hunk 2]
```diff
@@ -370,47 +422,50 @@ void SPlot::AddSWeight( RooAbsPdf* pdf, const RooArgList &yieldsTmp,
...
 
 
   // Set these parameters constant and store them so they can later
   // be set to not constant
-  std::vector<RooRealVar*> constVarHolder;
+  std::vector<RooAbsRealLValue*> constVarHolder;
 
   for(Int_t i = 0; i < constParameters->getSize(); i++)
+  {
+    RooAbsRealLValue* varTemp = dynamic_cast<RooAbsRealLValue*>( constParameters->at(i) );
```
Comment (eguiraud) [inline roofit/roostats/src/SPlot.cxx:444]: can these be anything else than `RooAbsRealLValue`s? (i.e. can this be a `static_cast`?)

Comment (hageboeck) [reply roofit/roostats/src/SPlot.cxx:444]: Yes, they can. It's being checked in the constructor. Thanks.

File: roofit/roostats/src/SPlot.cxx (hunk 1 above)
Comment (hageboeck) [inline roofit/roostats/src/SPlot.cxx:212]:
```suggestion
  // Add check that yieldsList contains all RooRealVar / RooAbsRealLValue
```

File: roofit/roostats/src/SPlot.cxx (hunk 2 above)
Comment (hageboeck) [inline roofit/roostats/src/SPlot.cxx:444]:
```suggestion
    RooAbsRealLValue* varTemp = static_cast<RooAbsRealLValue*>( constParameters->at(i) );
```

**Pull request merged**